lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -3153,6 +3283,322 @@ class SqlParser(
                 ErrorCode.PARSE_UNEXPECTED_TERM, PropertyValueMap()
             )
 
+    private fun List<Token>.parseOptionalMatchClause(child: ParseNode): ParseNode {
+        var rem = this
```

Comment:
variable could me made immutable

```suggestion
        val rem = this
```